sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeed.cs

```diff
@@ -245,18 +206,25 @@ internal DataFeedDetail GetDataFeedDetail()
         /// </summary>
         internal DataFeedDetailPatch GetPatchModel()
         {
-            DataFeedDetailPatch patch = DataSource.InstantiateDataFeedDetailPatch();
+            DataFeedDetailPatch patch = DataSource?.InstantiateDataFeedDetailPatch()
+                ?? new DataFeedDetailPatch();
```

Comment:
`DataSource` can be null now, since we don't require it in the constructor anymore. If it's not null, call `InstantiateDataFeedDetailPatch`, which will grab some extra information from the Data Source (information the user may be trying to update). If there's no Data Source, just create an empty instance and populate it below.